File: skill_sdk/tests/fixtures/ts-project/src/shop/models/Order.ts
import { OrderItem } from './OrderItem';
import { OrderStatus } from './OrderStatus';

/**
 * A customer order. Mirrors Order in the Java fixture.
 */
export class Order {
  private readonly createdAt: Date;
  private status: OrderStatus;
  private discountCode: string | null;

  constructor(
    private readonly orderId: string,
    private readonly userId: string,
    private readonly items: OrderItem[],
    discountCode: string | null
  ) {
    this.status = OrderStatus.PENDING;
    this.createdAt = new Date();
    this.discountCode = discountCode;
  }

  getOrderId(): string {
    return this.orderId;
  }

  getUserId(): string {
    return this.userId;
  }

  getItems(): OrderItem[] {
    return this.items;
  }

  getStatus(): OrderStatus {
    return this.status;
  }

  setStatus(status: OrderStatus): void {
    this.status = status;
  }

  getCreatedAt(): Date {
    return this.createdAt;
  }

  getDiscountCode(): string | null {
    return this.discountCode;
  }

  setDiscountCode(discountCode: string): void {
    this.discountCode = discountCode;
  }

  rawTotal(): number {
    let total = 0;
    for (const item of this.items) {
      total += item.subtotal();
    }
    return total;
  }

  applyDiscount(multiplier: number): number {
    return this.rawTotal() * multiplier;
  }
}

File: skill_sdk/tests/fixtures/java-project/src/main/java/com/fixture/shop/models/Order.java
package com.fixture.shop.models;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

/**
 * A customer order. Mirrors Order in shop/models.go.
 */
public class Order {

    private final String orderId;
    private final String userId;
    private final List<OrderItem> items;
    private OrderStatus status;
    private final LocalDateTime createdAt;
    private String discountCode;

    public Order(String orderId, String userId, List<OrderItem> items, String discountCode) {
        this.orderId = orderId;
        this.userId = userId;
        this.items = new ArrayList<>(items);
        this.status = OrderStatus.PENDING;
        this.createdAt = LocalDateTime.now();
        this.discountCode = discountCode;
    }

    public String getOrderId() { return orderId; }

    public String getUserId() { return userId; }

    public List<OrderItem> getItems() { return items; }

    public OrderStatus getStatus() { return status; }

    public void setStatus(OrderStatus status) { this.status = status; }

    public LocalDateTime getCreatedAt() { return createdAt; }

    public String getDiscountCode() { return discountCode; }

    public void setDiscountCode(String discountCode) { this.discountCode = discountCode; }

    public double rawTotal() {
        double total = 0;
        for (OrderItem item : items) {
            total += item.subtotal();
        }
        return total;
    }

    public double applyDiscount(double multiplier) {
        return rawTotal() * multiplier;
    }
}
